saleor/graphql/product/tests/test_product.py

```diff
@@ -3480,6 +3480,48 @@ def test_products_query_with_filter_stock_availability_as_staff(
     assert len(products) == 3
 
 
+def test_products_query_with_filter_stock_availability_including_reservations(
+    query_products_with_filter,
+    staff_api_client,
+    product_list,
+    order_line,
+    checkout_line,
+    permission_manage_products,
+    channel_USD,
+):
+
+    stocks = [product.variants.first().stocks.first() for product in product_list]
+    Allocation.objects.create(
+        order_line=order_line, stock=stocks[0], quantity_allocated=100
+    )
+    Reservation.objects.create(
+        checkout_line=checkout_line,
+        stock=stocks[1],
+        quantity_reserved=100,
+        reserved_until=timezone.now() + timedelta(minutes=5),
+    )
+    Allocation.objects.create(
+        order_line=order_line, stock=stocks[2], quantity_allocated=50
+    )
+    Reservation.objects.create(
```

Comment:
Maybe one of the reservations should be expired? Also, you could use `bulk_update` to reduce the number of queries.

Reply:
good point, thanks